src/openrct2/scripting/bindings/ride/ScRide.cpp

```diff
@@ -524,6 +524,72 @@ namespace OpenRCT2::Scripting
         return ride != nullptr ? ride->GetRideTypeDescriptor().LiftData.minimum_speed : 0;
     }
 
+    double ScRide::maxSpeed_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? (ride->max_speed * 9) / (1 << 18) : 0;
```

Comment:
Looks like this is copy-pasted? There should be a function available for this sort of thing.

Reply:
You're talking about the calculation for speed I presume. From what I can see, there's a couple of places that do have the same calculation like openrct2-ui/windows/InstallTrack.cpp but they're few and far in between. I think it's fine as is since the calc is small, but do you know of a util class I can refactor them into?

Reply:
The Util class where you already added a function will do. I want to avoid copypasting, even in this file alone it is already copied twice. The same applies to the other things here. (We are still suffering from bad code quality from the early days, so I don’t want to make it worse.)